app/models/utils/nav/customization_parser.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Utils
+  module Nav
+    # Parses the top menu customization setting
+    class CustomizationParser
+      attr_accessor :markdown_string
+
+      def initialize(markdown_string)
+        self.markdown_string = markdown_string
+      end
+
+      def parse
+        settings_topmenu = markdown_string || ""
+        settings_topmenu = settings_topmenu.scan(%r{\[([\w\- ]+)\]\s*\(([\w/\.:%&\?=\-_\',\+@]+)\)})
+        settings_topmenu.to_h
```

Comment:
I think we should be using the markdown parser here since it's markdown. See the Markdowner class. You can run it through that and then use `scan` to look for `<a ... href="..." ...>...</a>` and use that as your href and label. Then we're not duplicating Markdown parsing logic.

Sorry I didn't think of that sooner! This is why it's good to split things up into small bits!

Your specs shouldn't need to change at all though!

Reply:
Are you sure that's the best way to go?  When I originally wrote it, I thought about using the markdown parser, but decided just to use scan because (a) it would be as much or more trouble to parse the HTML, and (b) the parser would make things more complicated and would add a lot of runtime overhead.

Reply:
I guess I could go either way. Let's leave it as is.